src/components/scenes/Loans/LoanManageScene.tsx

```diff
@@ -158,17 +162,41 @@ export const LoanManageScene = (props: Props) => {
   const hardLtvRatio = '0.74'
   const isLtvExceeded =
     (actionOpType === 'loan-borrow' || actionOpType === 'loan-withdraw') &&
-    (zeroString(pendingCollateralsFiatValue) || gt(div(pendingDebtsFiatValue, pendingCollateralsFiatValue, DECIMAL_PRECISION), hardLtvRatio))
+    (lte(pendingCollateralsFiatValue, '0') || // Extra redundancy for withdrawing more than available collateral
+      gt(div(pendingDebtsFiatValue, pendingCollateralsFiatValue, DECIMAL_PRECISION), hardLtvRatio)) // Requested borrow or withdraw results in exceeding LTV
 
   // #endregion State
 
   // -----------------------------------------------------------------------------
   // #region Hooks
   // -----------------------------------------------------------------------------
 
+  // TODO: Full max button implementation and behavior
+  // Withdraw/repay ceilings
+  useAsyncEffect(async () => {
+    let availableNativeAmount: string | undefined
+
+    if (actionOpType === 'loan-withdraw') {
+      availableNativeAmount = collaterals.find(collateral => collateral.tokenId === selectedAsset.tokenId)?.nativeAmount
+    } else if (actionOpType === 'loan-repay') {
+      availableNativeAmount = debts.find(debt => debt.tokenId === selectedAsset.tokenId)?.nativeAmount
+    } else {
+      availableNativeAmount = undefined // Use uncapped actionNativeAmount
+    }
+
+    const cappedActionNativeAmount =
+      availableNativeAmount != null && (availableNativeAmount == null || gt(actionNativeAmount, availableNativeAmount))
```

Comment:
Isn't `availableNativeAmount != null && (availableNativeAmount == null...` redundant logic?

Reply:
Yeah, it is, but it was to get TS to stop complaining about `cappedActionNativeAmount` being potentially undefined.

I'll rework the logic here so it makes more sense